File: CS/WebSite/App_Code/MyHierarchicalView.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web.UI;

public class MyHierarchicalView : HierarchicalDataSourceView
{
    public override IHierarchicalEnumerable Select()
    {
        using (DataClassesDataContext context = new DataClassesDataContext())
        {
            List<MyHierarchyData> collection = context.Categories
                .Select(Category => new MyHierarchyData(Category.CategoryID.ToString(), null, Category.CategoryName)).ToList();
            return new MyHierarchicalCollection(collection);
        }
    }
}

File: VB/WebSite/App_Code/MyHierarchicalView.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web.UI

Public Class MyHierarchicalView
    Inherits HierarchicalDataSourceView

    Public Overrides Function [Select]() As IHierarchicalEnumerable
        Using context As New DataClassesDataContext()
            Dim collection As List(Of MyHierarchyData) = context.Categories.Select(Function(Category) New MyHierarchyData(Category.CategoryID.ToString(), Nothing, Category.CategoryName)).ToList()
            Return New MyHierarchicalCollection(collection)
        End Using
    End Function
End Class